packages/wrangler/src/api/pages/create-worker-bundle-contents.ts

```diff
@@ -73,7 +73,9 @@ function createWorkerBundleFormData(
 
 	// The upload API only accepts an empty string or no specified placement for the "off" mode.
 	const placement: CfPlacement | undefined =
-		config?.placement?.mode === "smart" ? { mode: "smart" } : undefined;
+		config?.placement?.mode === "smart"
+			? { mode: "smart", hint: config.placement.hint }
```

Comment:
Could this just be:

```suggestion
			? config.placement
```

Reply:
Just tried this but TypeScript can't narrow `config.placement` to `CfPlacement` here, so I think it's simpler to keep as-is

Reply:
OK. That's a shame. But fine as it is. Thanks for trying.